packages/firebase_dynamic_links/android/src/main/java/io/flutter/plugins/firebasedynamiclinks/FirebaseDynamicLinksPlugin.java

```diff
@@ -27,15 +31,30 @@
 import java.util.Map;
 
 /** FirebaseDynamicLinksPlugin */
-public class FirebaseDynamicLinksPlugin implements MethodCallHandler, NewIntentListener {
-  private final Registrar registrar;
-  private final MethodChannel channel;
+public class FirebaseDynamicLinksPlugin
+    implements FlutterPlugin, ActivityAware, MethodCallHandler, NewIntentListener {
+  private Registrar registrar;
+  private MethodChannel channel;
+  private ActivityPluginBinding activityBinding;
 
   private FirebaseDynamicLinksPlugin(Registrar registrar, MethodChannel channel) {
     this.registrar = registrar;
     this.channel = channel;
```

Comment:
You don't have to pass this in right? The less potentially dangling nulls there are (depending on the path you take), the less likely there will be accidental null pointer exceptions

Reply:
I should need it. The FirebaseDynamicLinks plugins makes callbacks to dart: https://github.com/FirebaseExtended/flutterfire/pull/1372/files#diff-3f6c424218a2662c8ddc4b322f30bf66R81. 

I could hold on to a `BinaryMessenger` and instantiate a `MethodChannel` every time we want to call `invokeMethod`, but I think this looks cleaner.